apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/rest/endpoint/UtilsController.java

```diff
@@ -0,0 +1,521 @@
+                    accountIdString);
+                log.warn(errorMessage);
+                return response.error(ApiErrors.INCORRECT_VALUE, "account", accountIdString).build();
+            }
+            dto.account = Long.toUnsignedString(accountId);
+            dto.accountRS = Convert2.rsAccount(blockchainConfig.getAccountPrefix(), accountId);
+            dto.accountLongId = String.valueOf(accountId);
+            log.debug("getRcConvert result: {}", dto);
+        } catch (RuntimeException e) {
+            String errorMessage = String.format("Incorrect account ID: value = %s",
+                accountIdString);
+            log.warn(errorMessage, e);
+            return response.error(ApiErrors.INCORRECT_PARAM_VALUE, "account", accountIdString).build();
+        }
+        return response.bind(dto).build();
+    }
+
+    @Path("/hash")
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(
+        summary = "The API hashes of passed data by using specified Hash algorithm",
+        description = "The API hashes of passed data by using specified Hash algorithm, data and if it's text of binary is passed with algorithm number",
+        tags = {"utility"},
+        responses = {
+            @ApiResponse(responseCode = "200", description = "Successful execution",
+                content = @Content(mediaType = "application/json",
+                    schema = @Schema(implementation = HashDto.class)))
+        })
+    public Response hashByAlgorithm(
+        @Parameter(name = "hashAlgorithm", description = "Valid Algorithm from available list", required = true,
+            schema = @Schema(implementation = HashFunction.class))
+            @QueryParam("hashAlgorithm") HashFunction hashAlgorithm,
+        @Parameter(name = "secretIsText", description = "false (default) is HEX string, true if data is plain Text")
+            @QueryParam("secretIsText") @DefaultValue("false") Boolean secretIsText,
+        @Parameter(name = "secret", description = "text or data to be hashed by selected algorithm", required = true)
+            @QueryParam("secret") @NotBlank String secret
+    ) {
+        ResponseBuilder response = ResponseBuilder.startTiming();
+        log.debug("Started hashByAlgorithm : \t 'hashAlgorithm' = {}", hashAlgorithm);
+        HashDto dto = new HashDto();
+        try {
+            byte[] secretAsByteArray = secretIsText != null && secretIsText ?
+                Convert.toBytes(secret) : Convert.parseHexString(secret);
+            dto.hash = Convert.toHexString(hashAlgorithm.hash(secretAsByteArray));
+            log.debug("hashByAlgorithm result: {}", dto);
+        } catch (RuntimeException e) {
+            String errorMessage = String.format("Error hashing by an algorithm = '%s', value = '%s'",
+                hashAlgorithm, secret);
+            log.warn(errorMessage, e);
+            return response.error(ApiErrors.INTERNAL_SERVER_EXCEPTION, errorMessage).build();
```

Comment:
I think that such error is misleading, maybe better to pick 'incorrect param' here since error may occurred only when secretAsByteArray is incorrect